lib/meta.nix

```diff
@@ -126,4 +126,18 @@ rec {
         lib.warn "getLicenseFromSpdxId: No license matches the given SPDX ID: ${licstr}"
         { shortName = licstr; }
       );
+
+  /* Get the path to the main program of a derivation with either
+     meta.mainProgram or pname or name
+
+     Type: getExe :: derivation -> string
+
+     Example:
+       getExe pkgs.hello
+       => "/nix/store/g124820p9hlv4lj8qplzxw1c44dxaw1k-hello-2.12/bin/hello"
+       getExe pkgs.mustache-go
+       => "/nix/store/am9ml4f4ywvivxnkiaqwr0hyxka1xjsf-mustache-go-1.3.0/bin/mustache"
+  */
+  getExe = x:
+    "${lib.getBin x}/bin/${x.meta.mainProgram or (lib.getName x)}";
```

Comment:
Having second thoughts about the `lib.getName x` part.

- A package *should* make its main program explicit. Explicit is good.
- When `/bin/${name}` does not exist, the error is delayed until runtime, and most likely the caller won't know why `/bin/foo`, which doesn't exist, was ever accessed. Especially if they didn't make the call to `getExe` themselves; for example when a NixOS option does it for them.

Should we make `meta.mainProgram` mandatory for `getExe`? I think we should!

Reply:
It'll have to be a warning for at least a release as this is already used widely outside of nixpkgs